Assets/MRTK/SDK/Features/UX/Scripts/Buttons/ButtonConfigHelper.cs

```diff
@@ -415,6 +414,168 @@ public void ForceRefresh()
+                    }
+
+                    string generatedIconSetPath = System.IO.Path.Combine(generatedIconSetFolder, generatedIconSetName + ".asset");
+                    targetIconSet = (ButtonIconSet)AssetDatabase.LoadAssetAtPath(generatedIconSetPath, typeof(ButtonIconSet));
+
+                    if (targetIconSet == null)
+                    {   // If the icon set doesn't already exist, duplicate the default
+                        ScriptableObject duplicateIconSet = Instantiate<ButtonIconSet>(defaultIconSet);
+                        duplicateIconSet.name = generatedIconSetName;
+
+                        AssetDatabase.CreateAsset(duplicateIconSet, generatedIconSetPath);
+                        AssetDatabase.SaveAssets();
+                        AssetDatabase.Refresh(ImportAssetOptions.ForceUpdate);
+
+                        targetIconSet = (ButtonIconSet)AssetDatabase.LoadAssetAtPath(generatedIconSetPath, typeof(ButtonIconSet));
+                        createdIconSet = true;
+                    }
+                }
+            }
+
+            bool selectIconSet = false;
+            if (createdIconSet)
+            {
+                selectIconSet = EditorUtility.DisplayDialog("Custom Icon Set Created", string.Format(customIconSetCreatedMessage, generatedIconSetFolder), "View Asset", "OK");
+            }
+
+            try
+            {
+                // Set the icon set to the custom generated icon set
+                iconSetProp.objectReferenceValue = targetIconSet;
+                // Add the custom icon to the custom set
+                targetIconSet.EditorAddCustomQuadIcon(targetQuadIcon);
+                // Reset changes to the quad renderer
+                iconQuadTextureProp.objectReferenceValue = targetQuadIcon;
+                configObject.ApplyModifiedProperties();
+
+                // If the custom material shader is different from the default material, don't alter the material
+                if (targetQuadMaterial.shader.name == defaultButtonQuadMaterial.shader.name)
+                {   // If the custom material shader is the same, revert any prefab overrides
+                    SerializedObject iconQuadRendererObject = new SerializedObject(iconQuadRenderer);
+                    SerializedProperty materialsProp = iconQuadRendererObject.FindProperty("m_Materials");
+                    PrefabUtility.RevertPropertyOverride(materialsProp, InteractionMode.AutomatedAction);
+                }
+            }
+            catch (System.Exception e)
+            {
+                Debug.LogError("There was an error when converting your custom button icon.");
+                Debug.LogError(e);
+                return;
+            }
+
+            EditorUtility.SetDirty(gameObject);
+
+            ForceRefresh();
+
+            if (selectIconSet)
+            {   // Wait for a moment to ensure the asset database has had a chance to refresh
+                await Task.Delay(100);
```

Comment:
Is there another way to ensure that the database is reloaded other than waiting for a magic number of seconds? Would also be good to not use a magic number here, at least describe what the variable means.

Reply:
Not that I can find. Refreshing the asset database is supposed to be enough, but if we don't wait for a few ms the subsequent Ping call behaves really strangely.

Reply:
I wonder what would happen if you refresh the asset database with ImportAssetOptions.ForceSynchronousImport? (Not sure how many assets you are reimporting.)

Reply:
That did it, thanks Cameron.